manufacturers/SenseCAP/models/S2100 - Generic/uplink_decoder.js

```diff
@@ -0,0 +1,202 @@
+        messages: []
+    };
+
+    // CRC check
+    if (!crc16Check(bytesString)) {
+        decoded['valid'] = false;
+        decoded['err'] = -1; // "crc check fail."
+        return {
+            data: decoded
+        };
+    }
+	
+	// Length Check
+    if (bytes.length!==11 && bytes.length!==15 && bytes.length!==22 && bytes.length!==32 && bytes.length!==33) {
+        decoded['valid'] = false;
+        decoded['err'] = -2; // "length check fail."
+        return {
+            data: decoded
+        };
+    }
+	
+	var measNum = bytes[1]
+	var meas_list = [];
+	if(bytes.length===11)
+	{
+		meas_list.push(((bytes[3] << 24) + (bytes[4] << 16) + (bytes[5] << 8) + bytes[6])/1000);
+		if(measNum===18)
+		{
+			meas_list.push(((bytes[7] << 24) + (bytes[8] << 16) + (bytes[9] << 8) + bytes[10])/1000);
+		}
+	}
+	else if(bytes.length===22)
+	{
+		meas_list.push(((bytes[3] << 24) + (bytes[4] << 16) + (bytes[5] << 8) + bytes[6])/1000);
+		meas_list.push(((bytes[7] << 24) + (bytes[8] << 16) + (bytes[9] << 8) + bytes[10])/1000);
+		meas_list.push(((bytes[14] << 24) + (bytes[15] << 16) + (bytes[16] << 8) + bytes[17])/1000);
+		var measNum4 = bytes[12]
+		if(measNum4===52)
+		{
+			meas_list.push(((bytes[18] << 24) + (bytes[19] << 16) + (bytes[20] << 8) + bytes[21])/1000);
+		}
+	}
+	else if(bytes.length===32)
+	{
+		meas_list.push(((bytes[3] << 24) + (bytes[4] << 16) + (bytes[5] << 8) + bytes[6])/1000);
+		meas_list.push(((bytes[7] << 24) + (bytes[8] << 16) + (bytes[9] << 8) + bytes[10])/1000);
+		meas_list.push(((bytes[13] << 24) + (bytes[14] << 16) + (bytes[15] << 8) + bytes[16])/1000);
+		meas_list.push(((bytes[17] << 24) + (bytes[18] << 16) + (bytes[19] << 8) + bytes[20])/1000);
+		meas_list.push(((bytes[24] << 24) + (bytes[25] << 16) + (bytes[26] << 8) + bytes[27])/1000);
+		var measNum6 = bytes[22]
+		if(measNum6===86)
+		{
+			meas_list.push(((bytes[28] << 24) + (bytes[29] << 16) + (bytes[30] << 8) + bytes[31])/1000);
+		}
+	}
+	else if(bytes.length===33)
+	{
+		meas_list.push(((bytes[3] << 24) + (bytes[4] << 16) + (bytes[5] << 8) + bytes[6])/1000);
+		meas_list.push(((bytes[7] << 24) + (bytes[8] << 16) + (bytes[9] << 8) + bytes[10])/1000);
+		meas_list.push(((bytes[14] << 24) + (bytes[15] << 16) + (bytes[16] << 8) + bytes[17])/1000);
+		meas_list.push(((bytes[18] << 24) + (bytes[19] << 16) + (bytes[20] << 8) + bytes[21])/1000);
+		meas_list.push(((bytes[25] << 24) + (bytes[26] << 16) + (bytes[27] << 8) + bytes[28])/1000);
+		var measNum6 = bytes[23]
+		if(measNum===86)
+		{
+			meas_list.push(((bytes[29] << 24) + (bytes[30] << 16) + (bytes[31] << 8) + bytes[32])/1000);
+		}
+	}
+	else if(bytes.length===15)
+	{
+		meas_list.push(bytes[1]);
+	}
+	
+	for(var i = 0; i<meas_list.length; i++)
+	{
+		if(bytes.length===15){
+			decoded.messages.push({
+				type: 'report_telemetry',
+				measurementId: 6,
+				measurementValue: meas_list[i]
+			});
+		}
+		else{
+			decoded.messages.push({
+				type: 'report_telemetry',
+				measurementId: i,
```

Comment:
It is not a good idea to make up our own measurementId, we should follow the official ones : https://sensecap-docs.seeed.cc/measurement_list.html
You can just take the S210X decoder provided by Seed studio(https://github.com/Seeed-Solution/SenseCAP-Decoder/blob/main/S210X/ChirpStack/SenseCAP_S210x_ChirpStackV3_Decoder.js), which should give those correct Ids as output, and add the ngsi-ld wrapper part.
That way we don't have to separate the decoder into a generic and leaf wetness ones (or even a S2104 one), we could have a single S210X decoder, since each measurement has its id it won't overlap

Reply:
S2100 and S210X don't use the same payload format, you can see on the Seed repository as well as they are separate . S2100 payload doesn't differentiate the measurements by ID but instead changes the format of the payload depending on the number measurements sent. 

The decoder from Seeed itself separates the measurements by the number e.g measurement 1 is id 1, measurement 2 is id 2 etc). Typically from our side we either adjust the decoder on the application server to indicate what id is corresponding to the measurement, e.g measurement 1 is changed to leaf wetness or we change it on the visualization side in our dashboards.

That is why I asked whether you want us to separate S2100 decoders by the sensors we will attach to them (and how you will identify them on your end) or whether I send you which S2100 EUIs are associated with which sensor and you manually change that on the application server.